### PR TITLE
[Add] transfer scripts and TaoDataset for qdtrack

### PR DESCRIPTION
**Pull request by Seerkfang**

### CODE REVIEW

File: tools/convert_datasets/mot/tao2coco.py [hunk 1]
```diff
@@ -0,0 +1,102 @@
+import argparse
```
Comment (GT9505) [inline tools/convert_datasets/mot/tao2coco.py:1]: `mkdir tao`, and remove into it

File: tools/convert_datasets/mot/merge_coco_with_lvis.py [hunk 2]
```diff
@@ -0,0 +1,99 @@
...
+import logging
+from pathlib import Path
+
+import numpy as np
+import pycocotools.mask as mask_util
+from pycocotools.coco import COCO
+from script_utils.common import common_setup
+from tqdm import tqdm
+
+ROOT = Path(__file__).resolve().parent.parent.parent
```
Comment (GT9505) [inline tools/convert_datasets/mot/merge_coco_with_lvis.py:13]: delete

File: tools/convert_datasets/mot/merge_coco_with_lvis.py [hunk 3]
```diff
@@ -0,0 +1,99 @@
...
+    for img_id, _ in coco.imgs.items():
+        if img_id in lvis.imgs:
+            coco_name = coco.imgs[img_id]['file_name']
+            lvis_name = lvis.imgs[img_id]['file_name']
+            assert coco_name in lvis_name
+        else:
+            logging.info(f'Image {img_id} in COCO, but not annotated in LVIS')
+
+    lvis_highest_id = max(x['id'] for x in lvis.anns.values())
+    ann_id_generator = itertools.count(lvis_highest_id + 1)
```
Comment (GT9505) [inline tools/convert_datasets/mot/merge_coco_with_lvis.py:66]: try `range`

File: tools/convert_datasets/mot/tao2coco.py [hunk 4]
```diff
@@ -0,0 +1,102 @@
...
+    out['tracks'] = raw['tracks'].copy()
+    out['info'] = raw['info'].copy()
+    out['licenses'] = raw['licenses'].copy()
+    out['categories'] = classes
+
+    for video in tqdm(raw['videos']):
+        img_infos = tao.vid_img_map[video['id']]
+        img_infos = sorted(img_infos, key=lambda x: x['frame_index'])
+        frame_range = img_infos[1]['frame_index'] - img_infos[0]['frame_index']
+        video['frame_range'] = frame_range
```
Comment (GT9505) [inline tools/convert_datasets/mot/tao2coco.py:65]: check these weird keys

File: mmtrack/datasets/tao_dataset.py [hunk 5]
```diff
@@ -0,0 +1,234 @@
...
+                If not specified, a temp file will be created. Default: None.
+
+        Returns:
+            tuple: (result_files, tmp_dir), result_files is a dict containing \
+                the json filepaths, tmp_dir is the temporal directory created \
+                for saving json files when jsonfile_prefix is not specified.
+        """
+        assert isinstance(results, dict), 'results must be a list'
+        assert 'track_results' in results
+        assert 'bbox_results' in results
```
Comment (GT9505) [inline mmtrack/datasets/tao_dataset.py:150]: `track_bboxes`
`det_bboxes`

File: mmtrack/datasets/tao_dataset.py [hunk 6]
```diff
@@ -0,0 +1,234 @@
...
+
+        track_results = self._track2json(results['track_results'])
+        result_files['track'] = f'{resfile_path}/tao_track.json'
+        mmcv.dump(track_results, result_files['track'])
+
+        return result_files, tmp_dir
+
+    def evaluate(self,
+                 results,
+                 metric=['track'],
```
Comment (GT9505) [inline mmtrack/datasets/tao_dataset.py:172]: `bbox`, 'track'

File: tools/convert_datasets/mot/merge_coco_with_lvis.py [hunk 7]
```diff
@@ -0,0 +1,99 @@
+import argparse
```
Comment (GT9505) [inline tools/convert_datasets/mot/merge_coco_with_lvis.py:1]: add comments to explain the script

**Review by GT9505 — COMMENTED** [review]
vscode python docstring generator

**Review by GT9505 — COMMENTED** [review]
1. complete the scripts including docstring
2. complete unittest
3. complete docs(*.md)

File: tools/convert_datasets/mot/merge_coco_with_lvis.py (hunk 3 above)
Comment (Seerkfang) [reply tools/convert_datasets/mot/merge_coco_with_lvis.py:66]: itertools.count is used to generate int after the highest id of lvis

File: tools/convert_datasets/mot/tao2coco.py (hunk 4 above)
Comment (Seerkfang) [reply tools/convert_datasets/mot/tao2coco.py:65]: after examination, every key has its meaning

**Pull request closed without merging**